src/HashMap.java

```diff
@@ -0,0 +1,160 @@
+            return;
+        }
+        checkOverflow();
+        key = key.toLowerCase();
+        int index = hash(key, this.tableSize);
+        index = checkIndex(key, index, this.keyTable, this.tableSize);
+        if (keyTable[index] != null) {
+            if (keyTable[index].compareTo(key) == 0) {
+                size -= 1;
+            }
+        }
+        keyTable[index] = key;
+        valueTable[index] = value;
+        size += 1;
+    }
+
+    @Override
+    public Integer get(String key) {
+        if (key == null) {
+            return null;
+        }
+        key = key.toLowerCase();
+        int index = hash(key, this.tableSize);
+        index = checkIndex(key, index, this.keyTable, this.tableSize);
+        if (keyTable[index] == null) {
+            return null;
+        }
+        return valueTable[index];
+    }
+
+    @Override
+    public void delete(String key) {
+        if (key == null) {
+            return;
+        }
+        key = key.toLowerCase();
+        int index = hash(key, this.tableSize);
+        index = checkIndex(key, index, this.keyTable, this.tableSize);
+        if (keyTable[index] != null) {
+            keyTable[index] = null;
+            valueTable[index] = 0;
+            size -= 1;
+        }
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public boolean contains(String key) {
+        if (key == null) {
+            return false;
+        }
+        key = key.toLowerCase();
+        int index = hash(key, this.tableSize);
+        index = checkIndex(key, index, this.keyTable, this.tableSize);
+        if (keyTable[index] == null) {
+            return false;
+        }
+        return true;
+    }
+
+    public void clear() {
+        primeNumberGenerator = new Prime(7);
+        tableSize = primeNumberGenerator.nextPrime();
+        keyTable = new String[tableSize];
+        valueTable = new int[tableSize];
+        size = 0;
+    }
+
+    public String[] keys() {
+        // TODO: added this for Watch support, need to return enumerator instead
+        String[] keys = new String[size];
+        int unUsedIndex = 0;
+
+        for (String key: this.keyTable) {
+            if (key != null) {
+                keys[unUsedIndex] = key;
+            }
+        }
+        return keys;
+    }
+
+    private int hash(String key, int tableSize) {
+        int sum = 0;
+
+        for (int i=0; i < key.length(); i++) {
+            sum += (int) key.charAt(i) * (i + 1);
+        }
+        return sum % tableSize;
+    }
+
+    private boolean indexTaken(String key, int index, String[] keyTable) {
+        if (keyTable[index] == null) {
+            return false;
+        }
+        else if (keyTable[index].compareTo(key) == 0) {
+            return false;
+        }
+        return true;
+    }
+
+    private int checkIndex(String key, int index, String[] keyTable, int tableSize) {
+        if (indexTaken(key, index, keyTable)) {
```

Comment:
Don't make this recursive. Use a while loop. While you haven't found a spot, keep checking. If you encounter the same spot you started from, you need to change your strategy (such as recheck the load factor, etc.)